lxd/container_lxc.go

```diff
@@ -1369,8 +1373,26 @@ func (c *containerLXC) initLXC() error {
 	return nil
 }
 
+// Initialize the storage pool for this container.
+func (c *containerLXC) initStoragePool() error {
+	if c.storagePool != "" {
+		return nil
+	}
+
+	_, rootDiskDevice := containerGetRootDiskDevice(c.localDevices)
+	if rootDiskDevice["pool"] == "" {
+		_, rootDiskDevice = containerGetRootDiskDevice(c.expandedDevices)
+	}
+	if rootDiskDevice["pool"] == "" {
+		return fmt.Errorf("No storage pool found in the containers devices.")
+	}
```

Comment:
- Right, what it does now, is that containerLXCCreate() parses the devices list to detect a valid storage pool. 
- init() now calls a function initStoragePool() that only looks at the database init also doesn't check for failure of initStoragePool(). It only warns when it failed to detect a valid storage pool